src/runtime-rs/crates/resource/src/volume/mod.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2019-2022 Alibaba Cloud
+// Copyright (c) 2019-2022 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+mod block_volume;
+mod default_volume;
+mod share_fs_volume;
+mod shm_volume;
+
+use std::{sync::Arc, vec::Vec};
+
+use anyhow::{Context, Result};
+use tokio::sync::RwLock;
+
+use crate::share_fs::ShareFs;
+
+pub trait Volume: Send + Sync {
+    fn get_volume_mount(&self) -> Result<Vec<oci::Mount>>;
+    fn get_storage(&self) -> Result<Vec<agent::Storage>>;
+    fn cleanup(&self) -> Result<()>;
+}
+
+#[derive(Default)]
+pub struct VolumeResourceInner {
+    volumes: Vec<Arc<dyn Volume>>,
+}
+
+impl VolumeResourceInner {
+    fn new() -> Self {
+        Self { volumes: vec![] }
+    }
+}
+
+#[derive(Default)]
+pub struct VolumeResource {
+    inner: Arc<RwLock<VolumeResourceInner>>,
+}
+
+impl VolumeResource {
+    pub fn new() -> Self {
+        Self {
+            inner: Arc::new(RwLock::new(VolumeResourceInner::new())),
+        }
+    }
+
+    pub async fn handler_volumes(
+        &self,
+        share_fs: &Option<Arc<dyn ShareFs>>,
+        cid: &str,
+        oci_mounts: &[oci::Mount],
+    ) -> Result<Vec<Arc<dyn Volume>>> {
+        let mut volumes: Vec<Arc<dyn Volume>> = vec![];
+        for m in oci_mounts {
+            let volume: Arc<dyn Volume> = match m {
```

Comment:
The match is redundant I think, use`if` only is enough

Reply:
Thanks for your review, change to use `if`.